core/src/main/java/io/questdb/cairo/mv/MatViewDefinition.java

```diff
@@ -0,0 +1,251 @@
+    ) {
+        this.matViewToken = matViewToken;
+        this.matViewSql = matViewSql;
+        this.baseTableName = baseTableName;
+        this.samplingInterval = samplingInterval;
+        this.samplingIntervalUnit = samplingIntervalUnit;
+        this.timeZone = timeZone;
+        this.timeZoneOffset = timeZoneOffset;
+
+        try {
+            this.timestampSampler = TimestampSamplerFactory.getInstance(
+                    samplingInterval,
+                    samplingIntervalUnit,
+                    0
+            );
+        } catch (SqlException e) {
+            throw CairoException.critical(0).put("invalid sampling interval and/or unit: ").put(samplingInterval)
+                    .put(", ").put(samplingIntervalUnit);
+        }
+
+        if (timeZone != null) {
+            try {
+                this.rules = Timestamps.getTimezoneRules(TimestampFormatUtils.EN_LOCALE, timeZone);
+            } catch (NumericException e) {
+                throw CairoException.critical(0).put("invalid timezone: ").put(timeZone);
+            }
+        } else {
+            this.rules = null;
+        }
+
+        if (timeZoneOffset != null) {
+            final long val = Timestamps.parseOffset(timeZoneOffset);
+            if (val == Numbers.LONG_NULL) {
+                throw CairoException.critical(0).put("invalid offset: ").put(timeZoneOffset);
+            }
+            this.fixedOffset = Numbers.decodeLowInt(val) * MINUTE_MICROS;
+        } else {
+            this.fixedOffset = 0;
+        }
+    }
+
+    public static void commitTo(@NotNull MetaFileWriter writer, @NotNull MatViewDefinition matViewDefinition) {
+        final AppendableBlock mem = writer.append();
+        writeTo(mem, matViewDefinition);
+        mem.commit(
+                MAT_VIEW_DEFINITION_FORMAT_MSG_TYPE,
+                MAT_VIEW_DEFINITION_FORMAT_MSG_VERSION,
+                MAT_VIEW_DEFINITION_FORMAT_FLAGS
+        );
+        writer.commit();
+    }
+
+    public static void writeTo(@NotNull AppendableBlock mem, @NotNull MatViewDefinition matViewDefinition) {
+        mem.putStr(matViewDefinition.getBaseTableName());
+        mem.putLong(matViewDefinition.getSamplingInterval());
+        mem.putChar(matViewDefinition.getSamplingIntervalUnit());
+        mem.putStr(matViewDefinition.getTimeZone());
+        mem.putStr(matViewDefinition.getTimeZoneOffset());
+        mem.putStr(matViewDefinition.getMatViewSql());
+    }
+
+    public static MatViewDefinition readFrom(@NotNull MetaFileReader reader, Path path, int rootLen, final TableToken matViewToken) {
+        path.trimTo(rootLen).concat(matViewToken.getDirName()).concat(MatViewDefinition.MAT_VIEW_DEFINITION_FILE_NAME);
+        reader.of(path.$());
+        MetaFileReader.BlockCursor cursor = reader.getCursor();
+        if (cursor.hasNext()) {
+            return loadMatViewDefinition(cursor.next(), matViewToken);
+        } else {
+            throw CairoException.critical(0)
```

Comment:
One of the points of having the file format with multiple blocks is to make it extendable and forward / backward compatible at the same time. 

When the unknown block is read from the file it means that the newer version of QuestDB wrote it with some extended informal the current version does not support. The reading should skip unknown blocks and keep reading known blocks.

This behavior when the file contains unknown block should be tested.